contracts/tests/verifier.test.ts

```diff
@@ -0,0 +1,376 @@
+      await expect(
+        verifier.getModels(testModelContentIds.length, 1),
+      ).to.be.revertedWith(revertMessage);
+    });
+  });
+
+  describe('getModelsByOwnerAddress', () => {
+    const testModelName = 'Test Model';
+    const testModelDescription = 'Description of test model.';
+
+    it('success', async () => {
+      const testModelContentIds: string[] = [
+        new Bytes32(
+          '0x1111111111111111111111111111111111111111111111111111111111111111',
+        ).toString(),
+        new Bytes32(
+          '0x1111111111111111111111111111111111111111111111111111111111111112',
+        ).toString(),
+        new Bytes32(
+          '0x1111111111111111111111111111111111111111111111111111111111111113',
+        ).toString(),
+      ];
+
+      testModelContentIds.forEach(async (testModelContentId) => {
+        await verifier.registerModel(
+          testModelContentId,
+          testModelName,
+          testModelDescription,
+        );
+      });
+
+      let models = await verifier.getModelsByOwnerAddress(ownerAddress, 0, 1);
+      expect(models.length).to.equal(1);
+      models = await verifier.getModels(0, 3);
+      expect(models.length).to.equal(3);
+      models = await verifier.getModels(0, 5);
+      expect(models.length).to.equal(3);
+      models = await verifier.getModels(1, 3);
+      expect(models.length).to.equal(2);
+    });
+
+    it('failure: model owner not found', async () => {
+      const revertMessage = 'model owner not found';
+      await expect(
+        verifier.getModelsByOwnerAddress(
+          '0x0000000000000000000000000000000000000000',
+          0,
+          1,
+        ),
+      ).to.be.revertedWith(revertMessage);
+    });
+
+    it('failure: invalid limit', async () => {
+      const testModelContentId: string = new Bytes32(
+        '0x1111111111111111111111111111111111111111111111111111111111111114',
+      ).toString();
+      await verifier.registerModel(
+        testModelContentId,
+        testModelName,
+        testModelDescription,
+      );
+
+      const revertMessage = 'limit must be > 0 and <= 30';
+      await expect(
+        verifier.getModelsByOwnerAddress(ownerAddress, 0, 0),
+      ).to.be.revertedWith(revertMessage);
+      await expect(
+        verifier.getModelsByOwnerAddress(ownerAddress, 0, 31),
+      ).to.be.revertedWith(revertMessage);
+    });
+
+    it('failure: invalid offset', async () => {
+      const testModelContentId: string = new Bytes32(
+        '0x1111111111111111111111111111111111111111111111111111111111111115',
+      ).toString();
+      await verifier.registerModel(
+        testModelContentId,
+        testModelName,
+        testModelDescription,
+      );
+
+      const revertMessage = 'offset must be < length of list of models';
+      await expect(
+        verifier.getModelsByOwnerAddress(ownerAddress, 5, 1),
+      ).to.be.revertedWith(revertMessage);
+    });
+  });
+
+  describe('updateModel', () => {
```

Comment:
This function (and `disableModel` function) are reverted if `msg.sender` doesn't match the `ownerAddress` field of the specified model (i.e. ensuring model developer A cannot modify models of other developers). How could we test such cases?

Reply:
- https://ethereum.stackexchange.com/questions/102610/change-transaction-msg-value-and-msg-sender-on-hardhat-test

Reply:
Fixed in 0533d915841d621a0f3f58397e629e7500413891.